bench/workgen/workgen.cxx

```diff
@@ -2701,6 +2888,75 @@ WorkloadRunner::run_all(WT_CONNECTION *conn)
+            std::cerr << "create_count needs to be greater than 0" << std::endl;
+            stopping = true;
+        } else if (options->create_target > 0 && options->create_trigger <= 0) {
+            std::cerr << "Need to specify a create_trigger when setting a create_target."
+                      << std::endl;
+            stopping = true;
+        } else if (options->create_trigger > 0 && options->create_target <= 0) {
+            std::cerr << "Need to specify a create_target when setting a create_trigger."
+                      << std::endl;
+            stopping = true;
+        } else if (options->create_trigger != 0 &&
+          options->create_target < options->create_trigger) {
+            std::cerr << "create_target should be greater than create_trigger." << std::endl;
+            stopping = true;
+        } else {
+            tableCreate = new WorkloadRunnerConnection();
+            tableCreate->runner = this;
+            tableCreate->connection = conn;
+
+            if ((ret = pthread_create(&tables_create_thandle, nullptr,
+                   thread_tables_create_workload, tableCreate)) != 0) {
+                std::cerr << "pthread_create failed err=" << ret << std::endl;
+                std::cerr << "Stopping table create thread." << std::endl;
+                (void)pthread_join(tables_create_thandle, &status);
+                delete tableCreate;
+                tableCreate = nullptr;
+                stopping = true;
+            }
+        }
+    }
+
+    // Start a thread to drop tables
+    pthread_t tables_drop_thandle;
+    WorkloadRunnerConnection *tableDrop = nullptr;
+    if (options->drop_interval > 0) {
+        if (options->drop_count < 1) {
+            std::cerr << "drop_count needs to be greater than 0" << std::endl;
+            stopping = true;
+        } else if (options->drop_target > 0 && options->drop_trigger <= 0) {
+            std::cerr << "Need to specify a drop_trigger when setting a drop_target." << std::endl;
+            stopping = true;
+        } else if (options->drop_trigger > 0 && options->drop_target <= 0) {
+            std::cerr << "Need to specify a drop_target when setting a drop_trigger." << std::endl;
+            stopping = true;
+        } else if (options->drop_target != 0 && options->drop_trigger < options->drop_target) {
+            std::cerr << "drop_trigger should be greater than drop_target." << std::endl;
+            stopping = true;
+        } else {
+            tableDrop = new WorkloadRunnerConnection();
+            tableDrop->runner = this;
+            tableDrop->connection = conn;
+
+            if ((ret = pthread_create(
+                   &tables_drop_thandle, nullptr, thread_tables_drop_workload, tableDrop)) != 0) {
+                std::cerr << "pthread_create failed err=" << ret << std::endl;
+                std::cerr << "Stopping table drop thread." << std::endl;
+                (void)pthread_join(tables_drop_thandle, &status);
```

Comment:
If the pthread_create failed, won't the handle be unusable (i.e. pthread_join may crash)?

Reply:
Fixed.